### PR TITLE
BN-1270 Improve forger finding probability / align P2P to Network simulator

### PR DESCRIPTION
**Pull request by o-aleksandrov-topl**
## Purpose
 Improve forger finding probability

## Approach
Inherit for neighbours only block providing reputation without a performance reputation

For any incoming connection do a measurement of performance reputation and save it to the appropriate Cold peer

A remote peer that provides absolutely new block (block source = 0) shall refresh noveltyReputation to remotePeerNoveltyInSlots value

Remove possible transition from warm to cold because of reputation

From remote peer request maximumWarmConnections instead of minimumWarmConnections 

SemiRandomSelectorColdToWarm => take double of reputation-based peers as Seq i.e. random peers proportion is going to 1/3 from 1/2

Reset peer reputation if peer moved to Cold state

ReputationRandomBasedSelectorWarmToHot => take “block providing reputation” + “perfromance reputation” + random warm peer instead of “overall repuation” + random warm peers

Change parameters:
  minimumHotConnections:  Int = 7,
  warmHostsUpdateEveryNBlock:           Double = 4.0,

  remotePeerNoveltyInExpectedBlocks:    Double = 4.0,
minimumPerformanceReputationPeers:    Int = 2

Request one new hot connection every remotePeerNoveltyInSlots slots

## Testing
Unit test + integration test

## Tickets
closes #BN-1270

### CODE REVIEW

File: networking/src/main/scala/co/topl/networking/fsnetwork/Notifier.scala [hunk 1]
```diff
@@ -137,11 +136,34 @@ object Notifier {
...
+    }
+  }
+
   private def finalizer[F[_]: Async: Logger](state: State[F]): F[Unit] =
     Logger[F].info("Stopping notification sending from Notifier actor") >>
     state.slotNotificationFiber.map(_.cancel).getOrElse(().pure[F]) >>
     state.networkQualityFiber.map(_.cancel).getOrElse(().pure[F]) >>
     state.warmHostsUpdateFiber.map(_.cancel).getOrElse(().pure[F]) >>
-    state.commonAncestorFiber.map(_.cancel).getOrElse(().pure[F])
+    state.commonAncestorFiber.map(_.cancel).getOrElse(().pure[F]) >>
```
Comment (SeanCheatham) [inline networking/src/main/scala/co/topl/networking/fsnetwork/Notifier.scala:166]: Not required, but this be simplified to `state.commonAncestorFiber.traverse(_.cancel)` I think

File: config/project/build.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+sbt.version=1.9.2
```
Comment (SeanCheatham) [inline config/project/build.properties:1]: Is this an accidental file?

**Review by SeanCheatham — APPROVED** [review]

File: config/project/build.properties (hunk 2 above)
Comment (o-aleksandrov-topl) [reply config/project/build.properties:1]: yes, will revert

File: networking/src/main/scala/co/topl/networking/fsnetwork/Notifier.scala (hunk 1 above)
Comment (o-aleksandrov-topl) [reply networking/src/main/scala/co/topl/networking/fsnetwork/Notifier.scala:166]: thanks, I will change

**Pull request merged**